core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateParameterConverter.java

```diff
@@ -110,6 +115,7 @@ private void initAzureEncryptionFromEnvironment(AzureInstanceTemplateV4Parameter
         if (diskEncryptionSetId != null) {
             LOGGER.info("Applying SSE with CMK for Azure managed disks as per environment.");
             AzureEncryptionV4Parameters encryption = new AzureEncryptionV4Parameters();
+            encryption.setKey(encryptionKeyUrl);
```

Comment:
what if `encryptionKeyUrl` is null? Will it work? I think null value should be handled here

Reply:
`encryptionKeyUrl` is pre-requisite for creation of ` diskEncryptionSetId`. Checking `diskEncryptionSetId != null` is sufficient to ensure that `encryptionKeyUrl` is non-null/present.

Reply:
I see, but this prerequisite (I guess it is a validation on API) is independent from this converter and anyone can change that prerequisite in the future but they won't know that they should change this logic too probably, so not checking for encryptionKeyUrl can be error prone in the future.